src/ar_node_utils.erl

```diff
@@ -135,42 +135,15 @@ calculate_reward_pool(OldPool, TXs, _RewardAddr, Proportion) ->
 %% @doc Calculates the portion of the rewardpool that the miner is entitled
 %% to for mining a block with a given recall. The proportion is based on the
 %% size of the recall block and the average data stored within the weave.
-calculate_proportion(RecallSize, WeaveSize, Height) when (Height == 0)->
-	% Genesis block.
-	calculate_proportion(
-		RecallSize,
-		WeaveSize,
-		1
-	);
-calculate_proportion(RecallSize, WeaveSize, Height) when (WeaveSize == 0)->
-	% No data stored in the weave.
-	calculate_proportion(
-		RecallSize,
-		1,
-		Height
-	);
-calculate_proportion(RecallSize, WeaveSize, Height) when RecallSize >= (WeaveSize/Height) ->
-	% Recall size is larger than the average data stored per block.
-	XRaw = ((Height * RecallSize) / WeaveSize) - 1,
-	X = min(XRaw, 1023),
-	max(
-		0.1,
-		(math:pow(2, X) / (math:pow(2, X) + 2))
-	);
-calculate_proportion(RecallSize, WeaveSize, Height) when RecallSize == 0 ->
-	% Recall block has no data txs, hence size of zero.
-	calculate_proportion(
-		1,
-		WeaveSize,
-		Height
-	);
+calculate_proportion(0, W, H) -> calculate_proportion(1, W, H);
```

Comment:
Why do we replace 0 recall size and weave size with 1? Shouldn't we in this case just return a minimal proportion right away?

Also, instead of replacing height 0 with 1 we can just divide by `H + 1` - I think, it is clearer this way.

Reply:
Disagree with your second point.

Agree with your first point.  Those first three branches could all return 0.1.  That will preserve behaviour (I think).